Builds/CMake/CMakeFuncs.cmake

```diff
@@ -502,6 +498,15 @@ macro(setup_build_boilerplate)
       APPEND
       PROPERTY COMPILE_DEFINITIONS
       $<$<OR:$<BOOL:${profile}>,$<CONFIG:Release>,$<CONFIG:ReleaseClassic>>:NDEBUG>)
+  else()
+    # CMAKE_CXX_FLAGS_RELEASE is created by CMake for most / all generators
+    #  with defaults including /DNDEBUG or -DNDEBUG, and that value is stored
+    #  in the cache. Override that locally so that the cache value will be
+    #  avaiable if "assert" is ever changed.
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_CXX_FLAGS_RELEASECLASSIC "${CMAKE_CXX_FLAGS_RELEASECLASSIC}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+    STRING(REGEX REPLACE "[-/][A-Z]*NDEBUG" "" CMAKE_C_FLAGS_RELEASECLASSIC "${CMAKE_C_FLAGS_RELEASECLASSIC}")
```

Comment:
That regex looks too permissive, is there a reason it's not something like `[-/]DNDEBUG`?

We could also use the list remove_item function for this: `LIST(REMOVE_ITEM CMAKE_CXX_FLAGS_RELEASE -DNDEBUG /DNDEBUG)` (assuming `/DNDEBUG` is what visual studio uses).